x-pack/plugins/apm/public/components/shared/Stacktrace/FrameHeading.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import React, { Fragment } from 'react';
+import styled from 'styled-components';
+import { colors, fontFamilyCode, px, units } from '../../../style/variables';
+
+const FileDetails = styled.div`
+  color: ${colors.gray3};
+  padding: ${px(units.half)};
+  font-family: ${fontFamilyCode};
+`;
+const LibraryFrameFileDetail = styled.span`
+  color: ${colors.gray2};
+`;
+const AppFrameFileDetail = styled.span`
+  font-weight: bold;
+  color: ${colors.black};
+`;
+
+interface StackframeLine {
+  number: number;
+}
+
+interface Stackframe {
+  filename: string;
+  function: string;
+  line: StackframeLine;
+}
+
+interface Props {
+  stackframe: Stackframe;
+  isLibraryFrame?: boolean;
+}
+
+const FrameHeading: React.SFC<Props> = ({
+  stackframe,
+  isLibraryFrame = false
+}) => {
+  const FileDetail = isLibraryFrame
+    ? LibraryFrameFileDetail
+    : AppFrameFileDetail;
+  const lineNumber: number = get(stackframe, 'line.number');
```

Comment:
Btw. we might be able to ditch `ts-optchain` and Lodash' `get` for idx: https://github.com/facebookincubator/idx